staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -483,20 +553,76 @@ func (o *AuditWebhookOptions) enabled() bool {
 	return o != nil && o.ConfigFile != ""
 }
 
-func (o *AuditWebhookOptions) applyTo(c *server.Config) error {
+func (o *AuditWebhookOptions) applyTo(c server.Config) (audit.Backend, error) {
 	if !o.enabled() {
-		return nil
+		return nil, nil
+	}
+	if c.AuditPolicyChecker == nil {
+		klog.V(2).Info("No audit policy file provided, no events will be recorded for webhook backend")
+		return nil, nil
 	}
 
 	groupVersion, _ := schema.ParseGroupVersion(o.GroupVersionString)
 	webhook, err := pluginwebhook.NewBackend(o.ConfigFile, groupVersion, o.InitialBackoff)
 	if err != nil {
-		return fmt.Errorf("initializing audit webhook: %v", err)
+		return nil, fmt.Errorf("initializing audit webhook: %v", err)
 	}
 	webhook = o.BatchOptions.wrapBackend(webhook)
-	webhook = o.TruncateOptions.wrapBackend(webhook, groupVersion)
-	c.AuditBackend = appendBackend(c.AuditBackend, webhook)
-	return nil
+	return webhook, nil
+}
+
+func (o *AuditDynamicOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&o.Enabled, "audit-dynamic-configuration", o.Enabled,
+		"Enables dynamic audit configuration")
```

Comment:
note in the help that it also requires enabling the alpha `DynamicAuditing` feature